### PR TITLE
Feature #1885 gh_projects

### LINKED ISSUE #1885: Add documentation for creating GitHub Projects to the Contributor's Guide
## Describe the Task ##
In the [GitHub Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) section of the Contributor's Guide:

1. Move section [6.2 Sequence of Events - Contributing Code](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html#sequence-of-events-contributing-code) to be section 6.3
2. Add a new section 6.2 Using GitHub Projects for Stages of the METplus Release Cycle.  In that section, write at least a few sentences to describe our use of GitHub Projects for the various [Stages of the METplus Release Cycle](https://metplus.readthedocs.io/en/latest/Release_Guide/index.html#stages-of-the-metplus-release-cycle).  Include the following directions for creating (copying) or updating a GitHub Project:
- copy/update the title and description
- add new columns to the default view
- define all the other views
- change from default private to public
- copy over the workflow automation settings
- add the METplus team with write permission plus any other admins
- go to each repo and "add" the new project to it

3.  While working on this page, fix the broken link for the text "best practices for collaboration on GitHub" in section 6.1.  The broken link is: https://metplus.readthedocs.io/en/latest/Contributors_Guide/docs.google.com/document/d/1DxsViogPdA0uObHgNx4YFKd4ClC-m9UFcX0rO-ZJTY0/edit?pli=1#heading=h.g4dgc8a6erna, whereas the link should be: https://docs.google.com/document/d/1DxsViogPdA0uObHgNx4YFKd4ClC-m9UFcX0rO-ZJTY0/edit?pli=1#heading=h.g4dgc8a6erna, and add a period at the end of the sentence.  Also change "WRF-Hydro team" to "formerly WRF-Hydro team" as James left RAL in January 2022 and is currently working in CPAESS but for USGS.

### Time Estimate ###
1 to 3 days of work

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
None

### Funding Source ###
TBD

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by JohnHalleyGotway**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Just wrote the docs as I created the new METplus-Analysis-5.1.0 Development project.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- @fisherhucar please review the newly created [METplus-Analysis-5.1.0 Development](https://github.com/orgs/dtcenter/projects/68) project contents.
- @jprestop please review the new GitHub workflow section [GitHub Projects to manage development](https://metplus.readthedocs.io/en/feature_1885_gh_projects/Contributors_Guide/github_workflow.html#github-projects-to-manage-development).
   - Julie, I did see the other proposed edits listed in the issue (fixing a broken link, changing the affiliation for James), but I couldn't actually find anything that needed to be fixed regarding that. So perhaps those changes have already been made?

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**
These are doc only updates.

- [x] Do these changes include sufficient testing updates? **[No]**
None needed.

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[Fri 12/2/22]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Development** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Contributors_Guide/github_workflow.rst
Comment (jprestop) [inline docs/Contributors_Guide/github_workflow.rst]:
```suggestion
        Under **More options** select a **Start on** date and set the default **Duration** as 6 weeks.
```

File: docs/Contributors_Guide/github_workflow.rst
Comment (jprestop) [inline docs/Contributors_Guide/github_workflow.rst]: This says "METplus Projects", but links to METviewer projects.  I am guessing that wasn't intended, but I'm not sure if you'd rather say "METviewer projects" or link to METplus or if you have no preference.  Please fix however you'd like.

**Review by jprestop — APPROVED** [review]
This is amazing!  Thank you!  I'd be happy to run through these steps in setting up a future project for the next release.  I made a couple of suggested changes:
1. Fix formatting for some bolded text
2. Fix either text or a link

Once those changes are made I approve this request.  I'll go ahead and approve now, instead of "Request changes" since they're minor changes and then you won't have to wait for me to approve in the future.

**Comment by JohnHalleyGotway** [comment]
- Added a new section named **GitHub Projects to manage support** and created a new [METplus Version 5.0 Support](https://github.com/orgs/dtcenter/projects/69) project.
- Just pushed changes to update the release guide slightly and add links to the contributor's guide.
- Not sure if I found all spots where changes are needed, but we can find/fix more later.
- Adding @georgemccabe as a reviewer to get his perspective.

**Comment by jprestop** [comment]
I went ahead and re-reviewed for the Release Notes changes.  It looks good to me @JohnHalleyGotway.  Thanks for realizing there were changes needed there as well.

File: docs/Contributors_Guide/github_workflow.rst
Comment (georgemccabe) [inline docs/Contributors_Guide/github_workflow.rst:148]: The METplus project boards have included another view that I find useful. This isn't mandatory to add.

```suggestion

      - View name **High/Blocker Not Required** shows all items labelled as *HIGH* or *BLOCKER* priority but
        not marked as required for the development or official release.
        Click on the 3 horizontal bars and define the filtering criteria as
        **is:open label:"priority: high","priority: blocker" -label:"required: FOR DEVELOPMENT RELEASE" -label:"required: FOR OFFICIAL RELEASE"**.
        Click **Save Changes**.

```

File: docs/Contributors_Guide/github_workflow.rst
Comment (georgemccabe) [inline docs/Contributors_Guide/github_workflow.rst]: If the suggestion below this one is applied, then the number here should be updated.

```suggestion
   - Click on the down arrow and select **Duplicate View**.  Do this 8 times and name/refine these views as follows.
```

**Review by georgemccabe — APPROVED** [review]
This content looks great, thank you for adding it. There is another label that is used for METplus that I find useful. The suggested changes include instructions to create it. If you don't think this is necessary, then feel free to ignore the suggestion. I approve either way.

**Comment by JohnHalleyGotway** [comment]
@jprestop and @georgemccabe, thanks for the reviews and suggestions. George, I added your recommended view.

I just removed @fisherhucar from the reviewers since 2 is certainly plenty.

Merging now.

**Pull request merged**
